common/src/python/preprocess/preprocessor.py

```diff
@@ -996,6 +887,118 @@ def is_existing_visit(self, *, input_record: Dict[str, Any]) -> bool:
 
         return False
 
+    def _check_clinical_forms(self, pp_context: PreprocessingContext) -> bool:
+        """Check at least one clinical form (UDS, BDS, MDS) exists.
+
+        Args:
+            pp_context: preprocessing context
+        Returns:
+            True if the preprocessing checks pass, false otherwise
+        """
+        # treat each form like a fake supplement module and see if any visits exist
+        for module in [
+            DefaultValues.UDS_MODULE,
+            DefaultValues.BDS_MODULE,
+            DefaultValues.MDS_MODULE,
+        ]:
+            supplement_module = SupplementModuleConfigs(
+                label=module, date_field=FieldNames.DATE_COLUMN, exact_match=False
+            )
+
+            # passes if at least one visit is found
+            if self.__get_supplement_visits(supplement_module, pp_context) is not None:
+                return True
+
+        # no clinical visits found; fail preprocessing check
+        self.__error_handler.write_module_error(
+            pp_context=pp_context, error_code=SysErrorCodes.CLINICAL_FORM_REQUIRED
+        )
+
+        return False
+
+    def _check_np_mlst_restrictions(self, pp_context: PreprocessingContext) -> bool:
+        """Check NP/MLST restrictions; compares NP form against most recent
+        MLST form.
+
+        Args:
+            pp_context: preprocessing context
+        Returns:
+            True if the preprocessing checks pass, false otherwise
+        """
+        if self.__module != DefaultValues.NP_MODULE:
+            raise PreprocessingException(
+                "Cannot evaluate NP/MLST preprocessing "
+                + f"checks for non-NP module: {self.__module}"
+            )
+
+        assert pp_context.subject_lbl, "pp_context.subject_lbl required"
+
+        # get most recent MLST form
+        all_mlst_forms = self.__forms_store.query_form_data(
+            subject_lbl=pp_context.subject_lbl,
+            module=DefaultValues.MLST_MODULE,
+            legacy=False,
+            search_col=FieldNames.DATE_COLUMN,
+            find_all=True,
+            extra_columns=["deathyr", "deathmo", "deathdy", "autopsy"],
+        )
+
```

Comment:
MLST form can be in legacy project, if not found in ingest project, query the legacy project